cmd/gen-go-sample/out.go

```diff
@@ -86,7 +86,12 @@ func writeOutputSpec(out OutputSpec, st *symTab, gen *generator) error {
 	}
 	if l := out.Loop; l != nil {
 		used++
-		err = writeLoop(l, st, gen)
+		err = errors.E(nil, "")
+		if l.Collection != "" {
```

Comment:
This might be the clearest, most central place to check that not both `.Collection` and `.Map` are specified, instead of the distributed checks you have elsewhere.